ksqldb-common/src/main/java/io/confluent/ksql/function/UdfIndex.java

```diff
@@ -307,8 +311,21 @@ public String toString() {
     }
 
     int compare(final Node other) {
```

Comment:
This either needs a javadoc or the function should be renamed to indicate how different functions are being compared. It's very unclear from the name.

Reply:
Javadocs seem to be used somewhat sparsely in the codebase, but I'd personally lean slightly towards that over changing the method signature that Almog gave the function in the first place. What's your preference? What makes the method unclear now through the name where before it might not have been?

Reply:
Adding a javadoc sounds good. Some files in the codebase have very thorough docs while others have basically none. Ideally we'd move towards adding more docs across the board.

> What makes the method unclear now through the name where before it might not have been?

The logic is more complicated now. Ideally we would've had a doc before too but previously the implementation was two lines of code so it wasn't too bad to read the implementation to understand what was happening. Now the implementation is a lot lengthier and harder to read. Adding a javadoc explaining what the method does in words will do future readers (and future you!) a favor.